config/gitleaks.toml

```diff
@@ -519,7 +519,10 @@ keywords = [
 ]
 # NOTE: Gitleaks >= v8.21.0 should use [[rules.allowlists] instead.
 [rules.allowlist]
-stopwords = [
+regexTarget = "match"
+regexes = [
+    '''(?i)(public[_.-]?(key|token)|(api|client)[_.-]?(version|id)|(secret|client)[_.-]?name|key[_.-]?word|monkey|donkey|keyboard|keying|keystone|rapid|capital)''',
+]stopwords = [
```

Comment:
Seems like the template is a little broken. 

This should fix it:
```diff
diff --git a/cmd/generate/config/rules/config.tmpl b/cmd/generate/config/rules/config.tmpl
index 9597b8f..2e79aa3 100644
--- a/cmd/generate/config/rules/config.tmpl
+++ b/cmd/generate/config/rules/config.tmpl
@@ -52,20 +52,18 @@ tags = [
 {{- if eq (len $rule.Allowlists) 1 }}{{ with index $rule.Allowlists 0}}{{ if or .Regexes .Paths .Commits .StopWords }}
 # NOTE: Gitleaks >= v8.21.0 should use [[rules.allowlists] instead.
 [rules.allowlist]
-{{ with .MatchCondition }}condition = "{{ . }}"
-{{ end -}}
-{{- with .Commits }}commits = [
+condition = "{{ .MatchCondition.String }}"
+{{- with .Commits -}}{{println}}commits = [
     {{ range $j, $commit := . }}"{{ $commit }}",{{ end }}
 ]{{ end }}
-{{- with .Paths }}paths = [
+{{- with .Paths }}{{println}}paths = [
     {{ range $j, $path := . }}'''{{ $path }}''',{{ end }}
 ]{{ end }}
-{{- with .RegexTarget }}regexTarget = "{{ . }}"
-{{ end -}}
-{{- with .Regexes }}regexes = [{{ range $i, $regex := . }}
+{{- if and .RegexTarget .Regexes }}{{println}}regexTarget = "{{ .RegexTarget }}"{{ end -}}
+{{- with .Regexes }}{{println}}regexes = [{{ range $i, $regex := . }}
     '''{{ $regex }}''',{{ end }}
 ]{{ end }}
-{{- with .StopWords }}stopwords = [{{ range $j, $stopword := . }}
+{{- with .StopWords }}{{println}}stopwords = [{{ range $j, $stopword := . }}
     "{{ $stopword }}",{{ end }}
 ]{{ end }}{{ end }}{{ end }}
 {{ else }}
@@ -84,7 +82,7 @@ tags = [
 {{- with $allowlist.Regexes }}regexes = [{{ range $i, $regex := . }}
     '''{{ $regex }}''',{{ end }}
 ]{{ end }}
-{{- with $allowlist.StopWords }}stopwords = [{{ range $j, $stopword := . }}
+{{ with $allowlist.StopWords }}stopwords = [{{ range $j, $stopword := . }}
     "{{ $stopword }}",{{ end }}
 ]{{ end }}{{ end }}{{ end }}
 {{ end }}{{ end }}
```

Reply:
Thanks! I have been trying to figure this out for the last hour, and could not make it work.

One thing: after applying this patch, I got the new line `condition = "OR" on 7 rules, which previously had no such setting. Is that intended?

e.g. 
```patch
diff --git a/config/gitleaks.toml b/config/gitleaks.toml
index 4d62cf3..a0370a1 100644
--- a/config/gitleaks.toml
+++ b/config/gitleaks.toml
@@ -157,7 +157,6 @@ keywords = [
 ]
 # NOTE: Gitleaks >= v8.21.0 should use [[rules.allowlists] instead.
 [rules.allowlist]
-condition = "OR"
 regexes = [
     '''.+EXAMPLE$''',
 ]
```

I tried to fix that by thinkering like this:
```patch
diff --git a/cmd/generate/config/rules/config.tmpl b/cmd/generate/config/rules/config.tmpl
index 2e79aa3..a0729d9 100644
--- a/cmd/generate/config/rules/config.tmpl
+++ b/cmd/generate/config/rules/config.tmpl
@@ -52,7 +52,7 @@ tags = [
 {{- if eq (len $rule.Allowlists) 1 }}{{ with index $rule.Allowlists 0}}{{ if or .Regexes .Paths .Commits .StopWords }}
 # NOTE: Gitleaks >= v8.21.0 should use [[rules.allowlists] instead.
 [rules.allowlist]
-condition = "{{ .MatchCondition.String }}"
+{{- with .MatchCondition }}{{println}}condition = "{{ .MatchCondition.String }}"{{ end }}
 {{- with .Commits -}}{{println}}commits = [
     {{ range $j, $commit := . }}"{{ $commit }}",{{ end }}
 ]{{ end }}
```

Now the toml file appears to look normally again.

Reply:
> One thing: after applying this patch, I got the new line `condition = "OR" on 7 rules, which previously had no such setting. Is that intended?

I hadn't really put much thought into it. The default value for it is `OR` so I suppose it doesn't need to be included.